### DISCUSSION TITLE
Does Immich overwrite files with similar filenames?

### DISCUSSION THREAD

**Discussion by briandipalma** [immich-app/immich#3324] [Q&A] [answered]
I've set my `Storage Template` value to `{{y}}/{{MM}}/{{y}}{{MM}}{{dd}}-{{hh}}{{mm}}{{ss}}`.
I was wondering what would happen if two files were uploaded that could produce the same name?

Would Immich overwrite the first one uploaded? Does it automatically add a sequence number to the filename?

**Comment by alextran1502** [comment, marked as answer]
It will add sequence number `+x` to the end of the filename

**Reply by GuoxiongXie** [reply]
Does this apply to photos uploaded through Immich CLI bulk upload? For example, let's say I already have a photo with file name "file1.jpg" in the app. If I use Immich CLI to upload a different photo with the same filename "file1.jpg": `immich upload file1.jpg`. Will a sequence number be added to the end of this second file to avoid collision? Thanks!

**Reply by jrasm91** [reply]
Correct. It applies to all files that are uploaded via any mechanism to immich.

**Reply by treverehrfurth** [reply]
How would one go about removing all these dups?
I had ran into an issue with an update and recreated my immich container, but now its backing up thousands of already backed up files instead of importing them

**Comment by xstaford** [comment]
After migrating files via storage template. {{y}}/{{yy}}{{MM}}{{dd}}_{{filetype}}-{{hh}}{{mm}}{{ss}} I have "same named" files with other register like date_time.JPG and date_time.jpg what is ok for Linux but not for a windows. So these files are not sync (back-up) with windows folder. By instructions it should add +i for files with the same date-time And it did it for most of it but not for all.. (for me it's about 100 files from 100k).

The reason is: storage template takes extention as it is in original file. 

Please make a way to change extention to normal register (not capital).

**Reply by fabb** [reply]
Has this been resolved?
